data/preprocess/util.py

```diff
@@ -0,0 +1,180 @@
+import numpy as np
+
+
+def store_map(symbol_map: dict, filename: str):
+    with open(filename, "w") as f:
+        for symbol, index in symbol_map.items():
+            f.write(f"{index}\t{symbol}\n")
+
+
+def process_raw_split_files(raw_split_files: dict, folder: str,  order_sop: bool = False):
```

Comment:
Perhaps: analyze_raw_splits?

Reply:
The result could be a custom class (instead of the tuple), which needs to be passed to process_splits. This way, we can change that later on, if needed